File: practice/algorithms/warmup/birthday_cake_candles/BirthdayCakeCandles.java
import java.util.*;

public class BirthdayCakeCandles {

    static int birthdayCakeCandles(int n, int[] ar) {
        int max = 0;
        for(int i = 0; i < ar.length; i++) {
            if(ar[i] > max) {
                max = ar[i];
            }
        }
        int count = 0;
        for(int i = 0; i < ar.length; i++) {
            if(ar[i] == max) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] ar = new int[n];
        for(int ar_i = 0; ar_i < n; ar_i++){
            ar[ar_i] = in.nextInt();
        }
        int result = birthdayCakeCandles(n, ar);
        System.out.println(result);
    }
}

File: practice/algorithms/warmup/birthday_cake_candles/BirthdayCakeCandles.scala
object BirthdayCakeCandles {

  def birthdayCakeCandles(n: Int, ar: Array[Int]): Int =  {
    val candleHeightGroups: Map[Int, Seq[Int]] = ar.toList.groupBy(i => i)
    val maxHeight = candleHeightGroups.keys.max
    candleHeightGroups.getOrElse(maxHeight, Seq.empty).size
  }

  def main(args: Array[String]) {
    val sc = new java.util.Scanner (System.in)
    val n = sc.nextInt()
    val ar = new Array[Int](n)
    for(ar_i <- 0 until n) {
      ar(ar_i) = sc.nextInt()
    }
    val result = birthdayCakeCandles(n, ar)
    println(result)
  }
}
